src/plugins/intel_gna/src/pre_post_process/data_storage_converter.hpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+
+namespace ov {
+namespace intel_gna {
+namespace pre_post_processing {
+
+class DataStorageConverter {
+public:
+    virtual void convert_matrix_fp32_to_int16_avx(int16_t* ptr_dst,
```

Comment:
I would remove avx from method name.

you could create separate implementation which do regular conversion without AVX and return this from factory so no checking would be needed.

let's discuss it.